lib/octokit/client/pages.rb

```diff
@@ -1,63 +1,89 @@
       # @see https://developer.github.com/v3/repos/pages/#get-information-about-a-pages-site
       def pages(repo, options = {})
         get "#{Repository.path repo}/pages", options
       end
 
-      # Get a specific Pages build by ID
+      # Enable a Pages site
+      #
+      # @param repo [Integer, String, Repository, Hash] A GitHub repository
+      # @param options [Object] :source source parameter
+      # @return <Sawyer::Resource> The new pages site
+      # @see https://developer.github.com/v3/repos/pages/#enable-a-pages-site
+      def enable_pages_site(repo, options = {})
+        opts = ensure_api_media_type(:pages_site, options)
+        post "#{Repository.path repo}/pages", opts
+      end
+
+      # Update information about a Pages site
+      #
+      # @param repo [Integer, String, Repository, Hash] A GitHub repository
+      # @param options [String] :cname Specify a custom domain for the repository. Sending a `null` value will remove the custom domain. For more about custom domains, see "[Using a custom domain with GitHub Pages](https://help.github.com/articles/using-a-custom-domain-with-github-pages/)."
+      # @param options [String] :source Update the source for the repository. Must include the branch name, and may optionally specify the subdirectory `/docs`. Possible values are `"gh-pages"`, `"master"`, and `"master /docs"`.
+      # @return <Boolean> True on success, false otherwise
+      # @see https://developer.github.com/v3/repos/pages/#update-information-about-a-pages-site
+      def update_pages_site(repo, options = {})
+        boolean_from_response :put, "#{Repository.path repo}/pages", options
+      end
+
+      # Disable a Pages site
+      #
+      # @param repo [Integer, String, Repository, Hash] A GitHub repository
+      # @return <Boolean> True on success, false otherwise
+      # @see https://developer.github.com/v3/repos/pages/#disable-a-pages-site
+      def disable_pages_site(repo, options = {})
+        opts = ensure_api_media_type(:pages_site, options)
+        boolean_from_response :delete, "#{Repository.path repo}/pages", opts
+      end
+
+      # Get a specific Pages build
       #
       # @param repo [Integer, String, Repository, Hash] A GitHub repository
-      # @param id [Integer, String] Build ID
-      # @return [Sawyer::Resource] Pages build information
-      # @see https://developer.github.com/v3/repos/pages/#list-a-specific-pages-build
-      # @example
-      #   Octokit.pages_build("github/developer.github.com", 5472601)
-      def pages_build(repo, id, options = {})
-        opts = ensure_api_media_type(:pages, options)
-        get "#{Repository.path repo}/pages/builds/#{id}", opts
+      # @param build_id [Integer] The ID of the build
+      # @return <Sawyer::Resource> A single pages build
+      # @see https://developer.github.com/v3/repos/pages/#get-a-specific-pages-build
+      def pages_build(repo, build_id, options = {})
```

Comment:
Naming seems a little odd, not sure if it's possible to make it more clear through auto generation.